packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -197,6 +199,13 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         this.toDispose.push(this.onTermDidClose);
         this.toDispose.push(this.onDidOpenEmitter);
 
+        this.toDispose.push(this.copyOnSelectionHandler);
+        this.toDispose.push(this.term.onSelectionChange(() => {
+            if (this.copyOnSelection) {
+                this.copyOnSelectionHandler.copy(this.term.getSelection());
```

Comment:
Try just `clipboardService.writeText(this.term.getSelection())`

Reply:
@azatsarynnyy @akosyakov It works only for localhost, but If I use remote host with theia `clipboardService.writeText` this service display notification `Access to the clipboard is denied. Check your browser's permission.`(for Chrome) on each terminal text selection. Such behavior I think for this case is agly. In mine implementation I wrote code, if user don't have permission to clipboard, we use document.executeCommand('copy'), this code works for remote host.

Reply:
then we should fix `ClipboardService` implementation with your approach, see https://github.com/eclipse-theia/theia/issues/6209

btw looking at the code installing a window listener for each terminal seems to be unnecessary, it can be installed just before executing copy and then immediately removed, so there are not n terminal times listeners, just one when copy happens

Reply:
@AndrienkoAleksandr thank you for the explanation!
If your remote host is on http but not https then it's expected.
It's because of `ClipboardService` uses `Async Clipboard API` which doesn't work for the pages loaded through http due to the security concerns. It supports https only.

I've updated https://github.com/eclipse-theia/theia/issues/6209.

Reply:
Does http really matter? For browser content one has to deploy behind https for security reasons, for Electron localhost is used.

Reply:
You're right! It works for localhost as well.